lib/tree.rb

```diff
@@ -61,7 +86,7 @@ def tree_data
       diameter_sum += apple.diameter
     end
 
-    avg_diameter = # It's up to you to calculate the average diameter for this harvest.
+    avg_diameter = basket.size > 0 ? diameter_sum*1.0/basket.size : 0
```

Comment:
I guess since it's not a separate method we know that we wait until the tree has apples to fill the basket so checking for a divide by 0 is overkill.